app/models/template.rb

```diff
@@ -31,6 +35,28 @@ def preview_host_collection
     Host.authorized(:view_hosts).order(:name).limit(100)
   end
 
+  def metadata
+    "<%#\n#{to_export(false).to_yaml.sub(/\A---$/, '').strip}\n%>\n\n"
```

Comment:
Shebang should be on the first line then, same as we have `<?xml` and others, `to_erb` keeps the first line because of this. The reason was that I find it more readable if there an empty line to separate. But I removed the "\n", I agree that splitting first line and the rest is even worse and one day we could fix templates so metadata is always first.